### PR TITLE
Make post page with comments on AJAX

### PR DESCRIPTION
**Pull request by AlexSukov**

### CODE REVIEW

**Comment by Magons** [comment]
@AlexSukov Can I review this PR?

**Comment by AlexSukov** [comment]
@Magons It works, but i want to ask something. I don't know how to get newly created on ajax comment id

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,6 +3,6 @@
 class ApplicationController < ActionController::Base
   self.responder = ApplicationResponder
   respond_to :html
-
+  respond_to :json
```
Comment (Magons) [inline app/controllers/application_controller.rb]: you can write this in one line `respond_to :html, :json`

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,58 +1,41 @@
 class CommentsController < ApplicationController
+  before_action :set_post
```
Comment (Magons) [inline app/controllers/comments_controller.rb:3]: include device authentication `before_action :authenticate_user!`

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -1,58 +1,41 @@
...
+    @comment = @post.comments.create(comment_params)
+    respond_with(@post, @comment)
   end
 
-  # PATCH/PUT /comments/1
-  # PATCH/PUT /comments/1.json
   def update
     @comment.update(comment_params)
-    respond_with(@comment)
+    respond_with(@comment, location: @post)
```
Comment (Magons) [inline app/controllers/comments_controller.rb]: `respond_with(@post, @comment)`

File: app/models/comment.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 class Comment < ApplicationRecord
-  belongs_to :user
+  validates :body, presence: true
```
Comment (Magons) [inline app/models/comment.rb]: validates should be under associations

**Comment by Magons** [comment]
for future bootstrap includes icons http://getbootstrap.com/components/#glyphicons

File: config/routes.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
   root to: 'posts#index'
-  resources :comments
-  resources :posts
+  resources :posts do
+    resources :comments
```
Comment (Magons) [inline config/routes.rb]: if u remove index and new action from controller u should delete them from routes

Comment (AlexSukov) [reply config/routes.rb]: @Magons what do you mean? i can't understand)

**Pull request merged**